src/api/math/float/tanh.rs

```diff
@@ -0,0 +1,29 @@
+//! Implements vertical (lane-wise) floating-point `tanh`.
+
+macro_rules! impl_math_float_tanh {
+    ([$elem_ty:ident; $elem_count:expr]: $id:ident | $test_tt:tt) => {
+        impl $id {
+            /// Tanh.
+            #[inline]
+            pub fn tanh(self) -> Self {
+                use crate::codegen::math::float::tanh::Tanh;
+                Tanh::tanh(self)
+            }
+        }
+
+        test_if!{
+            $test_tt:
+            paste::item! {
+                pub mod [<$id _math_tanh>] {
+                    use super::*;
+                    #[cfg_attr(not(target_arch = "wasm32"), test)] #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
+                    fn tanh() {
+                        let z = $id::splat(0 as $elem_ty);
+
+                        assert_eq!(z, z.tanh());
```

Comment:
This is failing with the following message:

```
running 7 tests
test v64::f32x2_math_tanh::tanh ... ok
test v256::f64x4_math_tanh::tanh ... FAILED
test v512::f32x16_math_tanh::tanh ... FAILED
test v128::f32x4_math_tanh::tanh ... FAILED
test v128::f64x2_math_tanh::tanh ... FAILED
test v256::f32x8_math_tanh::tanh ... FAILED
test v512::f64x8_math_tanh::tanh ... FAILED

failures:

---- v256::f64x4_math_tanh::tanh stdout ----
thread 'v256::f64x4_math_tanh::tanh' panicked at 'assertion failed: `(left == right)`
  left: `3`,
 right: `0`', src/v256.rs:66:1
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

---- v512::f32x16_math_tanh::tanh stdout ----
thread 'v512::f32x16_math_tanh::tanh' panicked at 'assertion failed: `(left == right)`
  left: `14`,
 right: `0`', src/v512.rs:60:1

---- v128::f32x4_math_tanh::tanh stdout ----
thread 'v128::f32x4_math_tanh::tanh' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `0`', src/v128.rs:43:1

---- v128::f64x2_math_tanh::tanh stdout ----
thread 'v128::f64x2_math_tanh::tanh' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`', src/v128.rs:60:1

---- v256::f32x8_math_tanh::tanh stdout ----
thread 'v256::f32x8_math_tanh::tanh' panicked at 'assertion failed: `(left == right)`
  left: `6`,
 right: `0`', src/v256.rs:49:1

---- v512::f64x8_math_tanh::tanh stdout ----
thread 'v512::f64x8_math_tanh::tanh' panicked at 'assertion failed: `(left == right)`
  left: `7`,
 right: `0`', src/v512.rs:79:1


failures:
    v128::f32x4_math_tanh::tanh
    v128::f64x2_math_tanh::tanh
    v256::f32x8_math_tanh::tanh
    v256::f64x4_math_tanh::tanh
    v512::f32x16_math_tanh::tanh
    v512::f64x8_math_tanh::tanh

test result: FAILED. 1 passed; 6 failed; 0 ignored; 0 measured; 4990 filtered out
```

The macros calling these tests are a bit opaque to me. What's confusing to me is that the RHS is 0 (as it should be), while the LHS, which is hardcoded, is not.

Reply:
It appears that something here is broken. The macros should be using the `Debug` implementation of the vector types when printing the result (which prints, e.g., `f32x4(1.0, 2.0, 3.0, 4.0)`), but this has clearly stopped working. I'll try to look into this.

Reply:
The error apparently came from the aligned writes/reads, which you have criticized in another comment.

Having changed those to unaligned made the errors go away.